File: SDKs/googleapis.com/sheets/v4/python/src/sdk/models/shared/scorecardchartspec.py
from dataclasses import dataclass, field
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from . import *

class ScorecardChartSpecAggregateTypeEnum(str, Enum):
    CHART_AGGREGATE_TYPE_UNSPECIFIED = "CHART_AGGREGATE_TYPE_UNSPECIFIED"
    AVERAGE = "AVERAGE"
    COUNT = "COUNT"
    MAX = "MAX"
    MEDIAN = "MEDIAN"
    MIN = "MIN"
    SUM = "SUM"

class ScorecardChartSpecNumberFormatSourceEnum(str, Enum):
    CHART_NUMBER_FORMAT_SOURCE_UNDEFINED = "CHART_NUMBER_FORMAT_SOURCE_UNDEFINED"
    FROM_DATA = "FROM_DATA"
    CUSTOM = "CUSTOM"


@dataclass_json
@dataclass
class ScorecardChartSpec:
    r"""ScorecardChartSpec
    A scorecard chart. Scorecard charts are used to highlight key performance indicators, known as KPIs, on the spreadsheet. A scorecard chart can represent things like total sales, average cost, or a top selling item. You can specify a single data value, or aggregate over a range of data. Percentage or absolute difference from a baseline value can be highlighted, like changes over time.
    """
    
    aggregate_type: Optional[ScorecardChartSpecAggregateTypeEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('aggregateType') }})
    baseline_value_data: Optional[ChartData] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('baselineValueData') }})
    baseline_value_format: Optional[BaselineValueFormat] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('baselineValueFormat') }})
    custom_format_options: Optional[ChartCustomNumberFormatOptions] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('customFormatOptions') }})
    key_value_data: Optional[ChartData] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('keyValueData') }})
    key_value_format: Optional[KeyValueFormat] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('keyValueFormat') }})
    number_format_source: Optional[ScorecardChartSpecNumberFormatSourceEnum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('numberFormatSource') }})
    scale_factor: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('scaleFactor') }})
    

File: SDKs/googleapis.com/sheets/v4/typescript/src/sdk/models/shared/scorecardchartspec.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { ChartData } from "./chartdata";
import { BaselineValueFormat } from "./baselinevalueformat";
import { ChartCustomNumberFormatOptions } from "./chartcustomnumberformatoptions";
import { KeyValueFormat } from "./keyvalueformat";


export enum ScorecardChartSpecAggregateTypeEnum {
    ChartAggregateTypeUnspecified = "CHART_AGGREGATE_TYPE_UNSPECIFIED",
    Average = "AVERAGE",
    Count = "COUNT",
    Max = "MAX",
    Median = "MEDIAN",
    Min = "MIN",
    Sum = "SUM"
}

export enum ScorecardChartSpecNumberFormatSourceEnum {
    ChartNumberFormatSourceUndefined = "CHART_NUMBER_FORMAT_SOURCE_UNDEFINED",
    FromData = "FROM_DATA",
    Custom = "CUSTOM"
}


// ScorecardChartSpec
/** 
 * A scorecard chart. Scorecard charts are used to highlight key performance indicators, known as KPIs, on the spreadsheet. A scorecard chart can represent things like total sales, average cost, or a top selling item. You can specify a single data value, or aggregate over a range of data. Percentage or absolute difference from a baseline value can be highlighted, like changes over time.
**/
export class ScorecardChartSpec extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=aggregateType" })
  aggregateType?: ScorecardChartSpecAggregateTypeEnum;

  @SpeakeasyMetadata({ data: "json, name=baselineValueData" })
  baselineValueData?: ChartData;

  @SpeakeasyMetadata({ data: "json, name=baselineValueFormat" })
  baselineValueFormat?: BaselineValueFormat;

  @SpeakeasyMetadata({ data: "json, name=customFormatOptions" })
  customFormatOptions?: ChartCustomNumberFormatOptions;

  @SpeakeasyMetadata({ data: "json, name=keyValueData" })
  keyValueData?: ChartData;

  @SpeakeasyMetadata({ data: "json, name=keyValueFormat" })
  keyValueFormat?: KeyValueFormat;

  @SpeakeasyMetadata({ data: "json, name=numberFormatSource" })
  numberFormatSource?: ScorecardChartSpecNumberFormatSourceEnum;

  @SpeakeasyMetadata({ data: "json, name=scaleFactor" })
  scaleFactor?: number;
}
